src/manifolds/ProductManifold.jl

```diff
@@ -578,6 +578,24 @@ function sharp!(M::ProductManifold, v::TFVector, x, w::CoTFVector)
     return v
 end
 
+function show(io::IO, M::ProductManifold)
+    length(M.manifolds) == 0 && return println(io, "ProductManifold()")
```

Comment:
Does this happen? Well it's not explicitly excluded, though it seems not that reasonable.

Reply:
Well if it can happen (it currently can), then the case should exist so the `show` doesn't error. Even simple calls like `ProductManifold() |> manifold_dimension` error, so I think we should just error of one tries to construct with no submanifolds, and then we don't need this.

Reply:
Sure, as long as we allow that to happen, it should have a show – I am also in favour of disallowing zero-element product manifolds.